homeassistant/components/recorder/migration.py

```diff
@@ -32,29 +29,20 @@ def migrate_schema(instance):
             )
 
         if current_version == SCHEMA_VERSION:
-            # Clean up if old migration left file
-            if os.path.isfile(progress_path):
-                _LOGGER.warning("Found existing migration file, cleaning up")
-                os.remove(instance.hass.config.path(PROGRESS_FILE))
             return
 
-        with open(progress_path, "w"):
-            pass
-
         _LOGGER.warning(
             "Database is about to upgrade. Schema version: %s", current_version
         )
 
-        try:
+        with instance.hass.timeout.freeze(DOMAIN):
             for version in range(current_version, SCHEMA_VERSION):
                 new_version = version + 1
                 _LOGGER.info("Upgrading recorder db schema to version %s", new_version)
                 _apply_update(instance.engine, new_version, current_version)
                 session.add(SchemaChanges(schema_version=new_version))
 
                 _LOGGER.info("Upgrade to version %s done", new_version)
-        finally:
-            os.remove(instance.hass.config.path(PROGRESS_FILE))
```

Comment:
Is this okay for the supervisor @pvizeli ?

Reply:
Yes, the current supervisor has disabled timeouts for new versions but that end in a blocking state. I come up with this solution because I don't want to hack somethings on the supervisor if we can fix it on the core. But we need to release the new supervisor before we release the new HA on stable (which we need do anyway for the new panel)